Sprint_2

apologet-insane opened the pull request.

apologet-insane committed:
Всё покрыто, если jacoco не врёт!

apologet-insane committed:
Delete .idea directory

egorvas commented on src/test/java/com/example/catGetFoodTest.java:
```diff
@@ -0,0 +1,27 @@
+    @Mock
+   Feline feline;
+   Predator predator;
+@Test
```
Ошибкой это не считаю, но рекомендую в будущем следить за отступами в коде.

egorvas commented on src/test/java/com/example/catMakeTest.java:
```diff
@@ -0,0 +1,21 @@
+
+    @Test
+    public void catMakeTest(){
+        Cat cat = new Cat(feline);
```
Зачем нужен этот тест, в нем нет никаких проверок. Такого быть не должно.

egorvas commented on src/test/java/com/example/catGetFoodTest.java:
```diff
@@ -0,0 +1,27 @@
+    public void catGetFoodTest () throws  Exception {
+    Cat cat = new Cat(feline);
+    Mockito.when (feline.eatMeat()).thenReturn(List.of("Животные", "Птицы", "Рыба"));
+    List<String> kushaet = cat.getFood();
```
я понимаю что это смешно, но старайся выбирать нормальные имена для переменных

egorvas commented on target/surefire-reports/com.example.catGetFoodTest.txt:
```diff
@@ -0,0 +1,4 @@
+-------------------------------------------------------------------------------
```
Обрати внимание, что в пул реквестах не должно быть ничего из папки target. Это твои локальный файлы и не нужны для работоспособности проекта, но при этом могут приватные данные. Это можно не править. Единственное исключения для отчета jacoco по условиям задания он должен быть приложен. Ты можешь добавить папку с ним в исключения Jacoco  или приложить папку с ним в основное дерево репозитория.

egorvas commented on src/test/java/com/example/lionGetKittensTest.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Test
+    public void getKittensTest() throws Exception {
+        Lion lion = new Lion("Самка", feline);
```
Ты во многих местах совершаешь одну и туже ошибку. Вместо того чтобы тестировать код приложения ты тестируешь свои же моки, которые задаешь. Моки нужны для того чтобы избавится от лишних зависимостей и упростить логику теста, а ты вместо этого тетсируешь сам мок. Если твой тест переписать, то можно его заменить на assertEquals (1, 1);
Найди в коде все похожие места и замени в них мок на реальный объект.

egorvas commented on src/test/java/com/example/lionDoesHaveManeParametrizedTest.java:
```diff
@@ -0,0 +1,39 @@
+        assertEquals(expectedHasMane, actualHasMane);
+    }
+
+    @Test(expected = Exception.class)
```
В случае если ожидаемым результатом теста является эксепшен я рекомендую проверять не только что он произошел, но и текст ошибки(если он есть). Я не считаю это ошибкой, можно не исправлять, но по желанию можешь попробовать добавить в тест проверку текста.

egorvas commented on src/test/java/com/example/lionDoesHasManeTest.java:
```diff
@@ -0,0 +1,47 @@
+
+
+       // this.exepctedHasMane = exepctedHasMane;
+Lion lion = new Lion ("Самец", feline);
```
Опять же беда с отступами, такой код очень сложно читать.

egorvas commented on src/test/java/com/example/felineGetKittensTest.java:
```diff
@@ -0,0 +1,37 @@
+    final private int expectedKittensCount;
+
+    public felineGetKittensTest(int a, int expectedKittensCount) {
+        this.a = a;
```
a это крайне плохое название переменной, нужно его исправить

egorvas commented on src/test/java/com/example/lionGetFoodTest.java:
```diff
@@ -0,0 +1,31 @@
+    @Test
+    public void getFoodTest() throws Exception {
+        Lion lion = new Lion("Самец", feline);
+       //если, например, в Lion изменится animalKind на "Травоядное", то тест отловит изменения
```
Так и проверяй то, что 
```
        Lion lion = new Lion("Самец", new Feline());
        assertEquals (lion.getFood(), List.of("Трава", "Различные растения"));
```
Зачем тебе лишние моки какие-то. Ты только делаешь свой тест сложнее и непонятнее.

egorvas commented on src/test/java/com/example/catGetSoundTest.java:
```diff
@@ -0,0 +1,24 @@
+package com.example;
```
Пожалуйста, сделай так, чтобы все тесты относящиеся к одному классу были в одном тестовом классе. Представь еслиб тебе нужно было бы писать тесты не на 3 класса, а на 100. Еслиб ты делал один класс на каждый тест, то тестовых файлов было бы сотни или даже тысячи.
Исключение делаем для параметризованных тестов их лучше выделять в отдельный класс.

apologet-insane committed:
Delete target directory

apologet-insane committed:
Работа над ошибками:
- переименовал переменные и навёл порядок в отступах;
- свёл тесты в классы;
- добавил assert на Null в тесте на создание Cat;
- обсудил с наставником моки для тестов на Lion: они нужны;
- не сумел проверить текст эксепшена, но научусь;

egorvas commented on src/test/java/com/example/catTest.java:
```diff
@@ -0,0 +1,41 @@
+    Feline feline = new Feline();
+
+    @Test
+    public void catMakeNotNullTest() {
```
Я не считаю это ошибкой, но и смысла в этом тесте не вижу. У тебя в других тестах вызывается создание этого объекта и проверять его отдельно смысла нет.

apologet-insane committed:
Тестовые классы теперь с большой буквы

apologet-insane commented:
Спасибо за дельные и чёткие замечания. И за принятие второй доработки!
